src/mn_setup.c

```diff
@@ -1185,52 +1221,64 @@ static void DrawInstructions()
         }
         else if (flags & S_RESET)
         {
-            s = "Restore defaults";
+            if (pad)
+            {
+                first = M_GetPlatformName(GAMEPAD_A);
+                second = M_GetPlatformName(GAMEPAD_B);
+            }
+            else
+            {
+                first = M_GetNameForKey(KEY_ENTER);
+                second = M_GetNameForKey(KEY_ESCAPE);
+            }
+
+            HelpString(&s, "[ %s ] OK, [ %s ] Cancel", first, second);
         }
     }
     else
     {
         if (flags & S_INPUT)
         {
-            switch (menu_input)
+            if (pad)
             {
-                case mouse_mode:
-                    s = "[ Del ] to clear";
-                    break;
-                case pad_mode:
-                    M_snprintf(joyb_buf, sizeof(joyb_buf),
-                               "[ %s ] to change, [ %s ] to clear",
-                               M_GetPlatformName(GAMEPAD_A),
-                               M_GetPlatformName(GAMEPAD_Y));
-                    s = joyb_buf;
-                    break;
-                default:
-                case key_mode:
-                    s = "[ Enter ] to change, [ Del ] to clear";
-                    break;
+                first = M_GetPlatformName(GAMEPAD_A);
+                second = M_GetPlatformName(GAMEPAD_Y);
+            }
+            else
+            {
+                first = M_GetNameForKey(KEY_ENTER);
+                second = M_GetNameForKey(KEY_DEL);
             }
+
+            HelpString(&s, "[ %s ] Change, [ %s ] Clear", first, second);
         }
         else if (flags & S_RESET)
         {
-            s = "Restore defaults";
+            if (pad)
+            {
+                first = M_GetPlatformName(GAMEPAD_A);
+            }
+            else
+            {
+                first = M_GetNameForKey(KEY_ENTER);
+            }
+
+            HelpString(&s, "[ %s ] Restore defaults", first);
         }
         else
         {
-            switch (menu_input)
+            if (pad)
             {
-                case pad_mode:
-                    M_snprintf(joyb_buf, sizeof(joyb_buf),
-                               "[ %s ] to change, [ %s ] to return",
-                               M_GetPlatformName(GAMEPAD_A),
-                               M_GetPlatformName(GAMEPAD_B));
-                    s = joyb_buf;
-                    break;
-                case key_mode:
-                    s = "[ Enter ] to change";
-                    break;
-                default:
-                    break;
+                first = M_GetPlatformName(GAMEPAD_A);
+                second = M_GetPlatformName(GAMEPAD_B);
             }
+            else
+            {
+                first = M_GetNameForKey(KEY_ENTER);
+                second = M_GetNameForKey(KEY_BACKSPACE);
+            }
+
+            HelpString(&s, "[ %s ] Change, [ %s ] Back", first, second);
```

Comment:
Is there a way to let the user know that the right mouse button is also "Back"? I don't think it's obvious.

Reply:
No, this is following what modern games are starting to do and the Kex port since mixed input devices is becoming a more common occurrence. Steam Deck or gamepad+gyro as mouse are two examples. Otherwise the prompts will rapidly cycle between different input types.

Reply:
Interesting thread: https://twitter.com/HilariousCow/status/1826680362376069214

Reply:
We can show this all the time, like `[mouse2/backspace] Back`. Right now it seems that `backspace` is the only way to exit the menu.

Hmm, I see the KEX port does the same thing. Okay, I guess it's obvious these days.